skylines/frontend/views/flight.py

```diff
@@ -448,26 +448,48 @@ def update():
     if 'pilot_id' in data:
         pilot_id = data['pilot_id']
 
-        if pilot_id is not None and not User.exists(id=pilot_id):
-            return jsonify(error='unknown-pilot'), 422
+        if pilot_id is not None:
 
-        if g.flight.pilot_id != pilot_id:
-            g.flight.pilot_id = pilot_id
+            if not User.exists(id=pilot_id):
+                return jsonify(error='unknown-pilot'), 422
 
-            # update club if pilot changed
-            if pilot_id is not None:
-                g.flight.club_id = User.get(pilot_id).club_id
+            pilot_club_id = User.get(pilot_id).club_id
+
+            if pilot_club_id != g.current_user.club_id:
```

Comment:
I think this is not designed for the case where neither the current user, nor the specified pilot are part of any club. In that case the `club_id` will be `None` in both cases.